Remove components that are marked disabled

leifmadsen opened the pull request:
Remove the k8s object when the component is marked 'enabled: false' so that components can be
brought up and down from the ServiceTelemetry CR.

csibbitt reviewed (approved):
Thanks! Noticed this yesterday, will test the fix sometime today, probably. I'm okay to merge it now as long as you've tried it

leifmadsen commented:
> Thanks! Noticed this yesterday, will test the fix sometime today, probably. I'm okay to merge it now as long as you've tried it

I tested one of the changes to make sure I didn't have syntax errors and that the `Alertmanager` object was actually removed. I'm going to do another local build and try flipping a few other switches on and off. Once I do that I will merge it, but I expect this to work.

Thanks for noticing it yesterday!

leifmadsen commented:
Did a quick test, and not everything removed like I expected, so I need to check if it was just a build/push procedure problem, or an actual issue in the Ansible.

leifmadsen commented:
I did a bunch of testing to make sure I could start up and remove various components. I got this result when setting everything to `enabled: false` which is what I expected:

```
oc get pods
NAME                                          READY   STATUS    RESTARTS   AGE
elastic-operator-746f86c956-hvjwb             1/1     Running   0          20h
ibm-auditlogging-operator-bb9876dc6-22hfp     1/1     Running   0          20h
interconnect-operator-5b474bdddc-mdwhb        1/1     Running   0          20h
prometheus-operator-688bb988cc-4f4zz          1/1     Running   0          20h
service-telemetry-operator-5d8d4f9676-9wxjs   2/2     Running   0          7m51s
smart-gateway-operator-58d6879cd-tjrtz        2/2     Running   0          20h
```

With the following `ServiceTelemetry` object configuration:

```
spec:
  alerting:
    enabled: false
  backends:
    events:
      elasticsearch:
        enabled: false
        storage:
          strategy: ephemeral
    metrics:
      prometheus:
        enabled: false
        storage:
          strategy: ephemeral
  graphing:
    enabled: false
  highAvailability:
    enabled: false
  transports:
    qdr:
      enabled: false
```

I updated the `ServiceTelemetry` object with `alerting.enabled: true` and that stood up the Alertmanager. I enabled `backends.metrics.prometheus.enabled: true` and that stood up Prometheus and the collectd/ceilometer Smart Gateways. The Smart Gateways were crash looping, then after a few moments I realize `transports.qdr.enabled: false` was still configured. Changed this to `true` resulted in a QDR setup, and SGs recovering.

Going to merge this down.
